src/main/java/net/minecraftforge/common/config/ConfigManager.java

```diff
@@ -161,10 +161,132 @@ public static void load(String modid, Config.Type type)
             }
         }
     }
+    
+    public static void updateConfig(String modid, Config.Type type)
+    {
+        FMLLog.fine("Attempting to update Configuration objects of mod %s for type %s", modid, type);
+        ClassLoader mcl = Loader.instance().getModClassLoader();
+        File configDir = Loader.instance().getConfigDir();
+        Multimap<Config.Type, ASMData> map = asm_data.get(modid);
+
+        if (map == null)
+            return;
+
+        for (ASMData targ : map.get(type))
+        {
+            try
+            {
+                Class<?> cls = Class.forName(targ.getClassName(), true, mcl);
+                String name = (String)targ.getAnnotationInfo().get("name");
+                if (name == null)
+                    name = modid;
+                String category = (String)targ.getAnnotationInfo().get("category");
+                if (category == null)
+                    category = "general";
+
+                File file = new File(configDir, name + ".cfg");
+
+                Configuration cfg = CONFIGS.get(file.getAbsolutePath());
+
+                updateConfig(cfg, cls, modid, type == Config.Type.INSTANCE, category);
+
+                cfg.save();
+
+            }
+            catch (Exception e)
+            {
+                FMLLog.log(Level.ERROR, e, "An error occurred trying to load a config for %s into %s", modid, targ.getClassName());
+                throw new LoaderException(e);
+            }
+        }
+    }
 
     // =======================================================
     //                    INTERNAL
     // =======================================================
+    static Configuration getConfiguration(String modid, String name) {
+        if(name == null || name.isEmpty())
+            name = modid;
+        File configDir = Loader.instance().getConfigDir();
+        File configFile = new File(configDir, name + ".cfg");
+        return CONFIGS.get(configFile.getAbsolutePath());
+    }
+    
+    private static void updateConfig(Configuration cfg, Class<?> cls, String modid, boolean isStatic, String category)
+    {
+        for (Field f : cls.getDeclaredFields())
+        {
+            if (!Modifier.isPublic(f.getModifiers()))
```

Comment:
Thew point isnt to get rid of getters/setters, the point is to get rid of nasty interaction with the Old-as-crap Configuration system we have right now and move to a more abstract system we can deal with in the backend ourselves.

IF we want to get fancy we can define getters and setters ourselves using a transformer so that it just 'magically' tracks the changes, but that's out of the scope of the current state.